Stacksplit

hamzah-hayat opened the pull request:
New Alt Click ability on stacks
Lets you split stacks into different size piles using a number

hamzah-hayat committed:
Added an AltClick function to split Stacks

Stacks now have an altclick function that lets you input how much you
would like to split the stack by

Also changed the regular click code by turning most of its code into a
proc, this reduced the amount of copy-pasting needed, as both the
regular and alt click on stacks do the same thing, except alt clicking
lets you input a number

hamzah-hayat committed:
Wrong var name

RemieRichards commented on code/game/objects/items/stacks/stack.dm:
```diff
@@ -225,19 +225,44 @@
+		else
+			change_stack(user,stackmaterial)
+			user << "<span class='notice'>You take [stackmaterial] sheets out of the stack</span>"
+	return
```
trailing return, remove it please.

optimumtact commented on code/game/objects/items/stacks/stack.dm:
```diff
@@ -225,19 +225,44 @@
+	. = F
+	F.copy_evidences(src)
+	user.put_in_hands(F)
+	src.add_fingerprint(user)
```
no need for src here

optimumtact commented on code/game/objects/items/stacks/stack.dm:
```diff
@@ -225,19 +225,44 @@
+	src.add_fingerprint(user)
+	F.add_fingerprint(user)
+	use(amount)
+	if (src && usr.machine==src)
```
at what point could this possibly disappear during this proc?

hamzah-hayat committed:
Edits based on feedback

-Removed trailing return
-removed uneeded if statement
-removed uneeded src as it should call the right method anyway

chickenboy10 commented:
I've made the changes based on your feedback, but im a bit wary of changing the code inside the change stack proc because it was written before I got to it and I dont want to break anything by making unnecessary changes.

hamzah-hayat committed:
Removed metal interact after splitting
